evalml/data_checks/target_leakage_data_check.py

```diff
@@ -69,11 +69,11 @@ def validate(self, X, y):
         Example:
             >>> import pandas as pd
             >>> X = pd.DataFrame({
-            ...    'leak': [10, 42, 31, 51, 61],
+            ...    'leak': [10, 44, 31, 51, 44],
             ...    'x': [42, 54, 12, 64, 12],
             ...    'y': [13, 5, 13, 74, 24],
             ... })
-            >>> y = pd.Series([10, 42, 31, 51, 40])
+            >>> y = pd.Series([10, 42, 31, 51, 42])
```

Comment:
I saw your comment about how woodwork will not calculate mutual information if one of the columns has all unique values. I guess that means that the mutual information is 0 in that case.

Does this mean we should update the `DefaultDataChecks` to use `method='pearson'` for regression? Columns can still be 95% correlated with the target if the target is all unique.